### PR TITLE
feat: api me with aad auth template

### PR DESCRIPTION
**Pull request by huimiu**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25150693

Add a new API message extension template to support AAD auth.

### CODE REVIEW

File: templates/csharp/api-message-extension-sso/infra/azure.bicep [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          name: 'AzureWebJobsStorage'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${functionStorage.name};AccountKey=${listKeys(functionStorage.id, functionStorage.apiVersion).keys[0].value};EndpointSuffix=${environment().suffixes.storage}' // Azure Functions internal setting
+        }
+        {
+          name: 'FUNCTIONS_EXTENSION_VERSION'
+          value: '~4' // Use Azure Functions runtime v4
+        }
+        {
+          name: 'FUNCTIONS_WORKER_RUNTIME'
+          value: 'node' // Set runtime to NodeJS
```
Comment (hund030) [inline templates/csharp/api-message-extension-sso/infra/azure.bicep:67]: Is the runtime setting correct?

File: templates/csharp/api-message-extension-sso/infra/azure.bicep [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          name: 'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${functionStorage.name};AccountKey=${listKeys(functionStorage.id, functionStorage.apiVersion).keys[0].value};EndpointSuffix=${environment().suffixes.storage}' // Azure Functions internal setting
+        }
+        {
+          name: 'WEBSITE_RUN_FROM_PACKAGE'
+          value: '1' // Run Azure Functions from a package file
+        }
+        {
+          name: 'WEBSITE_NODE_DEFAULT_VERSION'
+          value: '~18' // Set NodeJS version to 18.x
```
Comment (hund030) [inline templates/csharp/api-message-extension-sso/infra/azure.bicep:79]: Also here.

**Review by hund030 — COMMENTED** [review]
Is there any source code difference from no-auth api me template? Could we consider merge them into one template?

File: templates/csharp/api-message-extension-sso/infra/azure.bicep (hunk 1 above)
Comment (huimiu) [reply templates/csharp/api-message-extension-sso/infra/azure.bicep]: Updated, thanks!

File: templates/csharp/api-message-extension-sso/infra/azure.bicep (hunk 2 above)
Comment (huimiu) [reply templates/csharp/api-message-extension-sso/infra/azure.bicep]: Removed.

File: templates/js/api-message-extension-sso/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `teamsapp.yml`       | This is the main Teams Toolkit project file. The project file defines two primary things: Properties and configuration Stage definitions.                                                                                                               |
+| `teamsapp.local.yml` | This overrides `teamsapp.yml` with actions that enable local execution and debugging.                                                                                                                                                                   |
+| `aad.manifest.json`  | This file defines the configuration of Microsoft Entra app. This template will only provision [single tenant](https://learn.microsoft.com/azure/active-directory/develop/single-and-multi-tenant-apps#who-can-sign-in-to-your-app) Microsoft Entra app. |
+
+## How Azure Active Directory (AAD) Works
+
```
Comment (MuyangAmigo) [inline templates/js/api-message-extension-sso/README.md:53]: Azure Active Directory -> Microsoft Entra

Comment (huimiu) [reply templates/js/api-message-extension-sso/README.md:53]: Updated

**Comment by huimiu** [comment]
> Is there any source code difference from no-auth api me template? Could we consider merge them into one template?

The manifest, README, YML, and Bicep files differ from templates without auth. Do you think it might be complicated to generate different templates through variables? Should we merge them into one template?

**Review by hund030 — APPROVED** [review]

**Pull request merged**